pkg/proxy/healthcheck/healthcheck.go

```diff
@@ -197,7 +197,13 @@ var _ http.Handler = hcHandler{}
 
 func (h hcHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	h.hcs.lock.Lock()
-	count := h.hcs.services[h.name].endpoints
+	svc, ok := h.hcs.services[h.name]
```

Comment:
comment: if there is one more read operation. Then just have create another function for it.